libsupport/src/JSONTracer.cpp

```diff
@@ -20,6 +22,7 @@
 namespace {
 
 const std::regex kRssRegex("^Rss\\w+:\\s+([0-9]+) kB");
+std::mutex output_mutex;
```

Comment:
nit: `json_tracer_output_mutex` or something a bit more verbose.

Reply:
correct, it just adds the callback argument to support it

I think the naming should be fine tbh, it's in an unnamed namespace inside the JSONTracer file, this is also how it is currently named in ProgressLogger